### DISCUSSION TITLE
Unison Project concept and proposal

### DISCUSSION THREAD

**Discussion by hojberg** [unisonweb/unison#2606] [Ideas]
In the past few months I've on and off been tinkering with a proposal for the Unison Project concept. This will be a core concept of Unison and will be what Unison Share is primarily driven off of.

Read and explore the write-up and definitions on Unison Share: [Unison Project concept and proposal](https://share.unison-lang.org/latest/namespaces/hojberg/project)

There are certainly a lot of rough edges here still, but i'm keen on publicizing it and gathering some feedback even at this stage.

**Comment by stew** [comment]
regarding your questions about how we might implement security; we are planning on doing an oauth2 integration with some authc provider(s) perhaps github perhaps google. We will be using this oauth2 integration in order to validate the identity of people trying to access unison cloud and other adjacent services we might provide, I think it would fit here as well

**Reply by hojberg** [reply]
Yeah I think this is the right approach for sure. What we need to design further, is ownership and permissions on top of that

**Comment by stew** [comment]
A thing that I'm curious about is:

    Projects should have a guard-railed namespace structure

How might we do this? would we pre-create the structure for them? If they don't maintain our structure, what is going to go wrong and how?

**Reply by hojberg** [reply]
I'm thinking that at the point of creation of a project we setup the namespace structure. I think the major requirement will be where the README lives and where the project definition lives.

How we deal with things like PRs i'm less sure about still.
